### ISSUE TITLE
Push to gh-pages

### ISSUE THREAD

**Issue by jaypatel512** [ApiGen/ApiGen#549]
Hey guys, 

I am trying to use apigen for my documentation. I wish to run a command on travis-ci such that it automatically generates the docs from source folder, and push the result into gh-pages branch automatically.

**Comment by TomasVotruba** [comment]
Hey @jaypatel512 , what a right time :) . Just posted tutorial yesterday: https://twitter.com/Api_Gen/status/562969240555511809

**Comment by jaypatel512** [comment]
Thanks @TomasVotruba ! I will try to setup, and close the issue once done. Thanks.

**Comment by jaypatel512** [comment]
Worked like a charm. Thank you so much @TomasVotruba !

**Comment by TomasVotruba** [comment]
@jaypatel512 That's great! Do you have link to repo?

**Comment by jaypatel512** [comment]
Here it is, http://jaypatel512.github.io/PayPal-PHP-SDK/

However, I plan to make it for https://github.com/paypal/PayPal-PHP-SDK soon. Considering now that it is so easy to do it, I will talk to the team about this, and make it for PayPal-PHP-SDK

**Comment by TomasVotruba** [comment]
@jaypatel512 That's great news. If you need help with anything, just let me know.

Also if you have some ideas how improve the tutorial, just modify the wiki page.

**Comment by jaypatel512** [comment]
Definitely. I will.

**Comment by jaypatel512** [comment]
There you go ! Works like a charm : http://paypal.github.io/PayPal-PHP-SDK/

**Comment by TomasVotruba** [comment]
Awesome! Could you try 4.1-beta? There should be some few fixes: http://apigen.org

**Comment by jaypatel512** [comment]
Hey @TomasVotruba ! Btw, I just thought of a probable issue. Not entirely sure, but you could confirm that. What happens, if someone tries to run just sh generate-apigen.sh file from their computer locally, once they have checked out our source from github? Will he be able to push changes ?

**Comment by TomasVotruba** [comment]
I tried that:

1) username and email would be set to Travis
2) secret token would not work (no Travis to set it up)
3) it might work, if one has access to the repository

It might be better to add to script some check, so it could be run only by Travis.

**Comment by jaypatel512** [comment]
It would be nice to have a tiny check somewhere, where it sees if we are running it on travis and not on cli. Probably an environment variable that Travis sets that we could use to do it. Because, your blog is the only source right now, about integrating, it would be nice to update that a bit, to help others in future to not accidently do that. 

I will try too myself, and let you know if anything could be done to prevent such thing.

**Comment by TomasVotruba** [comment]
Great, thanks! Feel free to update the article and bash script in this repo as well.

**Comment by jaypatel512** [comment]
Hey @TomasVotruba ! We made 1-2 minor changes to your thread, and I thought of giving you back what we changed. May be helpful to change yours too. 

- In [travis.yml](https://github.com/paypal/PayPal-PHP-SDK/blob/master/.travis.yml#L22), we added two more checks that would determine if the branch is `master` only, and is not a `pull request` (as github recognize PRs to `master` as `master` branch). 
- we use `after_success` for running this script, as `after_script` might get executed on failed commit too.

Future:
- we were thinking of doing the same environment variable check again in the script itself, which would help us not get accidentally run in our local machine. (This may or may not be a good idea). For me, it changed my global user in github to travis, and caused me a bit of headache fixing all).

Anyway, I really appreciated your help on this. Our PHP SDK now has a nice looking [Github Page](http://paypal.github.io/PayPal-PHP-SDK/) with apigen source code, thanks to your post.
